### PR TITLE
dynamically load throttling tiers from tier xml

### PR DESCRIPTION
**Pull request by lahiru823**
Please review and commit

### CODE REVIEW

File: modules/apps/publisher/apis/v1/entitlement_api_router.jag [hunk 1]
```diff
@@ -11,6 +11,10 @@ require('/modules/publisher.js').exec(function(ctx) {
     var routeManager = new router.Router();
     var log= new Log();
 
+    routeManager.register('GET', 'publisher', '/publisher/api/entitlement/get/Tiers', function (context) {
```
Comment (rushmin) [inline modules/apps/publisher/apis/v1/entitlement_api_router.jag]: URL segments should start with a simple letter. 'Tiers' is wrong.

File: modules/apps/publisher/themes/appm/js/resource-add.js [hunk 2]
```diff
@@ -10,6 +13,21 @@ var RESOURCES = [
...
+        url: '/publisher/api/entitlement/get/Tiers',
+        type: 'GET',
+        contentType: 'application/json',
+        dataType: 'json',
+        async: false,
+        success: function (data) {
+            tiers = data;
+            throttlingTierControlBlock = drawThrottlingTiersDynamically();
+        },
+        error: function () {
```
Comment (dinusha-dilrukshi) [inline modules/apps/publisher/themes/appm/js/resource-add.js]: Log the error here using console.log().

File: modules/apps/publisher/themes/appm/js/resource-add.js [hunk 3]
```diff
@@ -1,3 +1,6 @@
+var tiers ={}; //contains Throttling tier details
```
Comment (dinusha-dilrukshi) [inline modules/apps/publisher/themes/appm/js/resource-add.js]: Add spaces between equal mark.

File: modules/apps/publisher/apis/v1/entitlement_api_router.jag [hunk 4]
```diff
@@ -160,8 +164,11 @@ require('/modules/publisher.js').exec(function(ctx) {
         }
     }
 
-
-
+    //Get Tier details
+    function getTiers()
+    {
```
Comment (dinusha-dilrukshi) [inline modules/apps/publisher/apis/v1/entitlement_api_router.jag]: Use one convention. Keep parenthesis just after function definition.

File: modules/apps/publisher/themes/appm/js/resource-edit.js [hunk 5]
```diff
@@ -1,6 +1,22 @@
+var tiers ={}; //contains Throttling tier details
```
Comment (dinusha-dilrukshi) [inline modules/apps/publisher/themes/appm/js/resource-edit.js]: Comments given in resource-add.js is applied to resource-edit.js as well.

File: modules/apps/publisher/apis/v1/entitlement_api_router.jag (hunk 1 above)
Comment (lahiru823) [inline modules/apps/publisher/apis/v1/entitlement_api_router.jag]: fixed with commit# https://github.com/lahiru823/carbon-appmgt/commit/89934533ddb0553867fd8962afdd4c906288d7ac

File: modules/apps/publisher/apis/v1/entitlement_api_router.jag (hunk 4 above)
Comment (lahiru823) [inline modules/apps/publisher/apis/v1/entitlement_api_router.jag]: fixed with commit# https://github.com/lahiru823/carbon-appmgt/commit/f2a0b0eac2018655f5d9ea1f1f7bf3cb55e3fbc6

File: modules/apps/publisher/themes/appm/js/resource-add.js (hunk 3 above)
Comment (lahiru823) [inline modules/apps/publisher/themes/appm/js/resource-add.js]: commit# https://github.com/lahiru823/carbon-appmgt/commit/5135b45ab49aaf1909b1096e0df0da0e640d9962

File: modules/apps/publisher/themes/appm/js/resource-add.js (hunk 2 above)
Comment (lahiru823) [inline modules/apps/publisher/themes/appm/js/resource-add.js]: commit# https://github.com/lahiru823/carbon-appmgt/commit/c116772263ddb1934e8541df26ba5e1a78021d66

File: modules/apps/publisher/themes/appm/js/resource-edit.js (hunk 5 above)
Comment (lahiru823) [inline modules/apps/publisher/themes/appm/js/resource-edit.js]: commit# https://github.com/lahiru823/carbon-appmgt/commit/5528ebdf86c75586c71a3a71f6369913d4c7bda4

**Pull request merged**